pkg/mtu/detect_linux.go

```diff
@@ -75,3 +75,33 @@ func autoDetect() (int, error) {
 
 	return EthernetMTU, nil
 }
+
+// getMTUFromIf finds the interface that holds the ip and returns its mtu
+func getMTUFromIf(ip net.IP) (int, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return 0, fmt.Errorf("Unable to list interfaces: %s", err)
```

Comment:
I think that we should use `errors.Wrap()` here to retain the original error properly through the stack. Similarly for other instances.